src/hotspot/share/gc/shenandoah/shenandoahMark.inline.hpp

```diff
@@ -57,8 +57,8 @@ void ShenandoahMark::dedup_string(oop obj, StringDedup::Requests* const req) {
   }
 }
 
-template <class T, StringDedupMode STRING_DEDUP>
-void ShenandoahMark::do_task(ShenandoahObjToScanQueue* q, T* cl, ShenandoahLiveData* live_data, StringDedup::Requests* const req, ShenandoahMarkTask* task) {
+template <class T, ShenandoahGenerationType GENERATION, StringDedupMode STRING_DEDUP>
```

Comment:
Consider extracting the GENERATION type from T (closure's type).

Reply:
This might be possible via the suggestion/comment further down at line 102. However, this might require a bit more refactoring of other code, which I'd prefer to do not here in this ticket, but separately.

I'll defer that to a separate PR for which I'll open a JBS in anticipation and link it here.

Reply:
Tracked in task ticket https://bugs.openjdk.org/browse/JDK-8314331